src/commands/lambda/function.ts

```diff
@@ -40,6 +40,46 @@ export interface InstrumentationSettings {
   tracingEnabled: boolean
 }
 
+const MAX_LAMBDA_STATE_CHECKS = 3
+
+/**
+ * Waits for n ms
+ * @param ms
+ * @returns
+ */
+const wait = (ms: number): Promise<void> => new Promise((res) => setTimeout(res, ms))
+
+/**
+ *
+ * @param config
+ * @param functionArn
+ * @param attempts
+ * @returns bool
+ */
+const checkLambdaState = async (
+  lambda: Lambda,
+  config: Lambda.FunctionConfiguration,
+  functionArn: string,
+  attempts = 0
+): Promise<boolean> => {
+  // TODO remove 1 Oct 2021 https://aws.amazon.com/blogs/compute/tracking-the-state-of-lambda-functions/
+  if (!config.State || !config.LastUpdateStatus) {
+    return true
+  }
+  if (config.LastUpdateStatus === 'Successful' && config.State === 'Active') {
+    return true
+  }
+  if (config.State === 'Pending' && attempts <= MAX_LAMBDA_STATE_CHECKS) {
+    await wait(2 ** attempts * 1000)
```

Comment:
At a glance I wasn't sure what the order of operations between ** and * is. Maybe add parenthesis? Or use  Math.pow over **.